core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -352,7 +352,7 @@ abstract class SparkFunSuite
         exp => {
           val parm = parameters.getOrElse(exp._1,
             throw new IllegalArgumentException("Missing parameter" + exp._1))
-          if (!exp._2.matches(parm)) {
+          if (!exp._2.matches(parm) && exp._2 != parm) {
```

Comment:
Since I am using `checkErrorMatchPVals` function, either I should:
- have this change so that we check regex **or** check if the exact value is the same or:
- from a variable which is the exact match to expected, create something like this: "parameter" -> ("^" + Pattern.quote(val) + "$").

I am using `checkErrorMatchPVals` in the first place, since I have some parameters that can only be checked with regex, but for some parameters I have the exact value.